### ISSUE TITLE
After spring-boot version update, some of the endpoints throw exception

### ISSUE THREAD

**Issue by markfarkas-camunda** [camunda/connectors#1869]
__Describe the Bug__

Calling the `/inbound` endpoint results in the exception below:

```
Servlet.service() for servlet [dispatcherServlet] in context with path [/aaaaaaaa-aaaa-4444-8888-877e6a418b53] threw exception [Request processing failed: java.lang.IllegalArgumentException: Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag.] with root cause
java.lang.IllegalArgumentException: Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag.
```

__Steps to Reproduce__

1. Call the /inbound endpoint
2. Observe the error response

__Expected Behavior__

Should work exactly as it did before the spring-boot version update.

__Environment__

N/A

__Additional context__

See https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#parameter-name-retention

**Comment by markfarkas-camunda** [comment]
Tested by QA on INT

**Comment by chillleader** [comment]
We are observing this issue again with the newly released 8.3.9.
